From 859d45bc59c2a1a22cc40bb2562b36ca3b89a8c6 Mon Sep 17 00:00:00 2001
From: "Eric J. Mislivec" <emislivec@nagios.com>
Date: Mon, 25 Aug 2014 09:12:17 -0500
Subject: [PATCH] runcmd: Fix build when we don't HAVE_SETENV.

A two character change (or move a '=' character...) to fix an
incredibly horrid mistake on my part that broke the build on Solaris,
OS X, and other systems that don't have setenv(). I didn't retest
without HAVE_SETENV after adding in the errror handling. I'm going to
frame this and put it on the wall next to the other smart things I've
done...
---
 lib/runcmd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/runcmd.c b/lib/runcmd.c
index a834708..4b164bc 100644
--- a/lib/runcmd.c
+++ b/lib/runcmd.c
@@ -587,9 +587,9 @@ static int runcmd_setenv(const char *name, const char *value) {
 	/* For Solaris and systems that don't have setenv().
 	 * This will leak memory, but in a "controlled" way, since the memory
 	 * should be freed when the child process exits. */
-	if (asprintf(&env_string, "%s=%s", name, value) = -1) return -1;
+	if (asprintf(&env_string, "%s=%s", name, value) == -1) return -1;
 	if (!env_string) {
-		errno == ENOMEM;
+		errno = ENOMEM;
 		return -1;
 	}
 	return putenv(env_string);
